pkg/workload/workload.go

```diff
@@ -497,13 +553,42 @@ func SetEvictedCondition(w *kueue.Workload, reason string, message string) {
 	apimeta.SetStatusCondition(&w.Status.Conditions, condition)
 }
 
+// PropagateResouceRequests synchronizes w.Status.ResourceRequests to
+// with info.TotalRequests and returns true if w was updated
+func PropagateResourceRequests(w *kueue.Workload, info *Info) bool {
+	if len(w.Status.ResourceRequests) == len(info.TotalRequests) {
+		match := true
+		for idx := range w.Status.ResourceRequests {
+			if w.Status.ResourceRequests[idx].Name != info.TotalRequests[idx].Name ||
+				!maps.Equal(w.Status.ResourceRequests[idx].ResourceRequest, info.TotalRequests[idx].Requests.ToResourceList()) {
+				match = false
+				break
+			}
+		}
+		if match {
+			return false
+		}
+	}
+
+	res := make([]kueue.PodSetRequest, len(info.TotalRequests))
+	for idx := range info.TotalRequests {
+		res[idx].Name = info.TotalRequests[idx].Name
+		res[idx].ResourceRequest = info.TotalRequests[idx].Requests.ToResourceList()
+	}
+	w.Status.ResourceRequests = res
+	return true
+}
+
 // AdmissionStatusPatch creates a new object based on the input workload that contains
 // the admission and related conditions. The object can be used in Server-Side-Apply.
 // If strict is true, resourceVersion will be part of the patch.
 func AdmissionStatusPatch(w *kueue.Workload, strict bool) *kueue.Workload {
```

Comment:
Should this function automatically add a clearing of `ResourceRequests` by setting it to an length 0 array of `Admission` is non-null?

Reply:
I went ahead and added logic to clear `Status.DesiredResources` when the workload is Admitted.    Results in a pretty clear status I think as observed via `kubectl describe`. 

While the job is waiting:
```
Status:
  Conditions:
    Last Transition Time:  2024-10-01T01:25:34Z
    Message:               couldn't assign flavors to pod set main: insufficient unused quota for cpu in flavor default-flavor, 1 more needed
    Observed Generation:   1
    Reason:                Pending
    Status:                False
    Type:                  QuotaReserved
  Desired Resources:
    Name:  main
    Resources:
      Cpu:  1
Events:
  Type    Reason   Age   From             Message
  ----    ------   ----  ----             -------
  Normal  Pending  41s   kueue-admission  couldn't assign flavors to pod set main: insufficient unused quota for cpu in flavor default-flavor, 1 more needed
```

Then after the job  is admitted:
```
Status:
  Admission:
    Cluster Queue:  cluster-queue
    Pod Set Assignments:
      Count:  1
      Flavors:
        Cpu:  default-flavor
      Name:   main
      Resource Usage:
        Cpu:  1
  Conditions:
    Last Transition Time:  2024-10-01T01:26:22Z
    Message:               Quota reserved in ClusterQueue cluster-queue
    Observed Generation:   1
    Reason:                QuotaReserved
    Status:                True
    Type:                  QuotaReserved
    Last Transition Time:  2024-10-01T01:26:22Z
    Message:               The workload is admitted
    Observed Generation:   1
    Reason:                Admitted
    Status:                True
    Type:                  Admitted
Events:
  Type    Reason         Age   From             Message
  ----    ------         ----  ----             -------
  Normal  Pending        61s   kueue-admission  couldn't assign flavors to pod set main: insufficient unused quota for cpu in flavor default-flavor, 1 more needed
  Normal  QuotaReserved  13s   kueue-admission  Quota reserved in ClusterQueue cluster-queue, wait time since queued was 48s
  Normal  Admitted       13s   kueue-admission  Admitted by ClusterQueue cluster-queue, wait time since reservation was 0s
```